### DISCUSSION TITLE
how the training/fitting happens when both x_train and y_train are of different sizes?

### DISCUSSION THREAD

**Discussion by teddcp2** [feature-engine/feature_engine#367] [Q&A] [answered]
hi all,

I have a doubt. It's like I have divided the data by calling train_test_split of SKLEARN. Then I am removing the outliers using percentile method for training set only. 

So now my x_train is reduced compared to what it had. 

But here I have question i.e. even if my y_train still has the same old size and x_train has a different size, My model is getting trained. Won't it be wrong to let the model fitting the data ?

If not, how the model is accompanied in the change?

**Comment by solegalli** [comment, marked as answer]
Yes, if you remove outliers from the train set using the Outlier trimmer, then you need to remove from y_train those observations as well.
You can do so my looking at the X_train and y_train indeces.

**Reply by teddcp2** [reply]
Hi @solegalli  , Thanks for answering. Yep, I understood your point. Thanks a lot.

Just one last question. I was going through your [cross-validation example](https://github.com/solegalli/feature-engineering-for-machine-learning/blob/master/Section-13-Putting-it-altogether/13.03-Assembling-pipeline-with-crossvalidation.ipynb) and let's say i want to do `SKLEARN CV with OutlierTrimmer`.

It won't work right? Because of the size mismatch between training and test set.

To do a CV with OutlierTrimmer, manually need to write a code to iterate over the data n-fold times and split data , remove outliers and handle missing values for training set, finally measure the score for test set.

Let me know if my above approach is correct or not. Thanks in advance :)

But with Winsorizer, SKLEARN CV will work.right ? As there wont be any size mismatch issues.

**Reply by solegalli** [reply]
that is correct
